docs/ftbbl.ipynb

```diff
@@ -0,0 +1,493 @@
+   "id": "148b727e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "run_id = raw_results.shared_runtime_info.run_id\n",
+    "run_id"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "60fd9675",
+   "metadata": {},
+   "source": [
+    "## Step 3: Analysis and Plotting\n",
+    "\n",
+    "Finally, we can analyze one or more datasets and generate plots. Since we've decoupled problem generation and execution from this step you can slice and dice your data any way you want. Usually, analysis routines will use the accompanying `analysis` module for helper function and do much of the `pd.DataFrame` and `matplotlib` munging interactively in a Jupyter notebook. One of the plots from `plots.ipynb` is reproduced here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "17c2a8c4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import recirq.otoc.loschmidt.tilted_square_lattice.analysis as analysis\n",
+    "import cirq_google as cg\n",
+    "import pandas as pd\n",
+    "\n",
+    "raw_results = cg.ExecutableGroupResultFilesystemRecord.from_json(run_id=run_id, base_data_dir=base_data_dir)\\\n",
+    "                    .load(base_data_dir=base_data_dir)\n",
+    "df = analysis.loschmidt_results_to_dataframe(raw_results)\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4d83dac4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "vs_depth_df, vs_depth_gb_cols = analysis.agg_vs_macrocycle_depth(df)\n",
+    "fit_df, exp_ansatz = analysis.fit_vs_macrocycle_depth(df)\n",
+    "total_df = pd.merge(vs_depth_df, fit_df, on=vs_depth_gb_cols)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ff91c006",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from matplotlib import pyplot as plt\n",
+    "\n",
+    "colors = plt.get_cmap('tab10')\n",
+    "\n",
+    "for i, row in total_df.iterrows():\n",
+    "    plt.errorbar(\n",
+    "        x=row['macrocycle_depth'],\n",
+    "        y=row['success_probability_mean'],\n",
+    "        yerr=row['success_probability_std'],\n",
+    "        marker='o', capsize=5, ls='',\n",
+    "        color=colors(i),\n",
+    "        label=f'{row[\"width\"]}x{row[\"height\"]} ({row[\"n_qubits\"]}q) {row[\"processor_str\"]}; f={row[\"f\"]:.3f}'\n",
+    "    )\n",
+    "    \n",
+    "    xx = np.linspace(np.min(row['macrocycle_depth']), np.max(row['macrocycle_depth']))\n",
+    "    yy = exp_ansatz(xx, a=row['a'], f=row['f'])\n",
+    "    plt.plot(xx, yy, ls='--', color=colors(i))\n",
+    "    \n",
+    "plt.legend(loc='best')\n",
+    "plt.yscale('log')\n",
+    "plt.xlabel('Macrocycle Depth')\n",
+    "plt.ylabel('Success Probability')\n",
+    "plt.tight_layout()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9b4b73a3",
+   "metadata": {},
+   "source": [
+    "## Appendix: `QuantumExecutable` and `ExecutableSpec`\n",
```

Comment:
Even after reading this, I am still a little confused about this.  I think a diagram (as mentioned above) would really help.

Reply:
working on some diagrams for this, stay tuned.